python/tank/context.py

```diff
@@ -113,23 +113,78 @@ def __str__(self):
         return ctx_name
 
     def __eq__(self, other):
+        """
+        Test if this Context instance is equal to the other Context instance
+                        
+        :param other:   The other Context instance to compare with
+        :returns:       True if self == other, False otherwise
+        """
         if not isinstance(other, Context):
             return NotImplemented
 
-        equal = True
-        equal &= (self.project == other.project)
-        equal &= (self.entity == other.entity)
-        equal &= (self.step == other.step)
-        equal &= (self.task == other.task)
-        equal &= (self.user == other.user)
-        equal &= (self.additional_entities == other.additional_entities)
-        return equal
+        if not self._entity_dicts_eq(self.project, other.project):
+            return False
+        
+        if not self._entity_dicts_eq(self.entity, other.entity):
+            return False
+        
+        if not self._entity_dicts_eq(self.step, other.step):
+            return False
+        
+        if not self._entity_dicts_eq(self.task, other.task):
+            return False
+        
+        if not self._entity_dicts_eq(self.user, other.user):
+            return False
+
+        # compare additional entities - note that currently this assumes the additional entity 
+        # lists are in exactly the same order for both self and other.        
+        if isinstance(self.additional_entities, list) and isinstance(other.additional_entities, list):
```

Comment:
From my understanding, additional_entities should always be lists based on all the code that depends on it everywhere.

If having something other than a list can break the internal logic of the class and all its clients, I suggest you validate that additional_entities is an array in __init__ and raise an exception there instead of having context comparison fail because of bad input.